Sources/_MatchingEngine/Regex/AST/MatchingOptions.swift

```diff
@@ -79,6 +93,74 @@ extension AST {
       self.minusLoc = minusLoc
       self.removing = removing
     }
+    
+    public func options(merging optionSet: MatchingOptionSet = []) -> MatchingOptionSet {
+      var result = optionSet
+      for opt in adding {
```

Comment:
`MatchingOptionSequence` also has a bit that indicates whether it clears out all the prior matching options except the ones it's adding, currently it's stored as `caretLoc != nil` (really it should have a proper API though). Would it be worth adding support for that feature here so we can handle e.g `(?^i)`?

Reply:
Oh yeah, this is the right place for that! I'll add that.

Reply:
Done — I also implemented the `U` option, which has an effect outside of a consumer closure, to see the effects in testing better.